src/backend/utils/text-parser.js

```diff
@@ -1,20 +1,28 @@
-const puppeteer = require('puppeteer');
+const { Cluster } = require('puppeteer-cluster');
 
-module.exports = async function(htmlFragment) {
-  let browser;
-  try {
-    browser = await puppeteer.launch();
-    const page = await browser.newPage();
+let isInitialized = false;
+let cluster;
 
-    await page.setContent(htmlFragment, {
-      waitUntil: 'domcontentloaded',
+module.exports.parse = async function(htmlFragment) {
+  if (!isInitialized) {
+    cluster = await Cluster.launch({
+      concurrency: Cluster.CONCURRENCY_CONTEXT,
+      maxConcurrency: 2,
```

Comment:
Probably worth throwing this in the env, so it can be set to something else.  I'd also suggest leaving it at the default for dev, 1.

Reply:
i'm running into a weird bug here. if I do `maxCurrency: 1`, then the tests will pass. However, if I do `maxConcurrency: process.env.TEXT_PARSER_MAX_CONCURRENCY` (I have the variable setup in `.env`, then running `npm test -- test/text-parser.test.js` will just hang, and after a bit it says:
```
Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.Error: Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.

      at mapper (node_modules/jest-jasmine2/build/queueRunner.js:25:45)
```
followed by
```
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

I think this is due to an underlying issue somewhere... the tests didn't pass on travis CI either with the same sort of errors suggesting jest didn't cleanly due to async processes hanging.

Reply:
On CI you'll need `process.env.TEXT_PARSER_MAX_CONCURRENCY || 1`, since we don't have an environment there.

Reply:
oh, I see. thank you for pointing it out.